### ISSUE TITLE
Protocol requesting pos-or-keyword param accepts pos-only param

### ISSUE THREAD

**Issue by JelleZijlstra** [microsoft/pyright#2652]
```python
from typing import Protocol


class _Writer(Protocol):
    def write(self, s: str) -> object:
        pass


class Writer:
    def write(self, s, /):
        pass


def make_writer(w: _Writer):
    pass


make_writer(Writer())
```

This should be disallowed because the protocol would allow calling `w.write(s=...)`, but the `Writer` class only allows positional calls.

This example was inspired by the `csv` stubs. Fixing this would likely lead to a lot of false positives on code using `csv.writer`. I'll submit a PR to typeshed shortly to fix the protocol there.

I'm on Pylance v2021.11.2.

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
In advance of your change to `_csv.pyi`, I've manually incorporated this change to the typeshed stubs shipped with pyright. Without this change, I'm seeing many false positive errors.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.192, which I just published. It will also be included in the next release of pylance.
